lib/acts_as_approvable/approval.rb

```diff
@@ -7,13 +7,17 @@ class Approval < ActiveRecord::Base
 
   validates_presence_of  :item
   validates_inclusion_of :event, :in => %w(create update destroy)
-  validates_numericality_of :state, :greater_than_or_equal_to => 0, :less_than => STATES.length
+  validate :valid_state
 
   serialize :object
   serialize :original
 
   before_save :able_to_save?
 
+  def valid_state
+    attributes['state'] >= 0 && attributes['state'] < STATES.length
```

Comment:
Validations should add to the error messages rather than rely on returning a result here.  For example, you'd typically see something like:

```ruby
def valid_state
  if attributes['state'] < 0 || attributes['state'] >= STATES.length
    errors.add :state, "must be greater than or equal to 0 and less than #{STATES.length}"
  end
end
```

You'll also need to consider what happens when `state` is `nil`.